advanced-solidus/images-and-image-processing.md

```diff
@@ -0,0 +1,196 @@
+
+{% code title="app/overrides/amazing_store/spree/active_storage_adapter/attachment/new_variant.rb" %}
+```ruby
+module AmazingStore
+  module Spree
+    module ActiveStorageAdapter
+      module Attachment
+        module NewVariant
+
+          def variant(style = nil)
+            size = style_to_size(style)
+            @attachment.variant(
+              resize_and_pad: size.append(background: "white"),
+              format: :png,
+              saver: { quality: 40 },
+              rotate: 180,
+              strip: true
+            ).processed
+
+          end
+          ::Spree::ActiveStorageAdapter::Attachment.prepend self
+        end
+      end
+    end
+  end
+end
+```
+{% endcode %}
+
+Take note that the method resize_and_pad expected an array input which is given by `size`, but the [documentation](https://github.com/janko/image_processing/blob/master/doc/minimagick.md#resize_and_pad) also states that the method expects the user to define what color the padded background will be. Without modifying any other methods, we could utilize the append method on our existing array to provide the required parameters for the method. For more information on methods that can be utilized to customize your image library, please refer to [MiniMagick's documentation](https://github.com/minimagick/minimagick).
+
+### Paperclip
+
+Paperclip utilizes [ImageMagick](https://imagemagick.org/index.php) similarly to the way ActiveStorage utilizes MiniMagick to modify its images. MiniMagick is just a wrapper for the more robust ImageMagick which comes with a [multitude of options](https://imagemagick.org/script/command-line-options.php) to process your images but has less compiled option methods such as `resize_and_pad`.
+We can still accomplish the scenario previously mentioned in the ActiveStorage guide above, however, the approach will be slightly different. Instead of modifying an existing class, we will implement our Paperclip attachment module. Despite only needed to modify a few lines of code to process images, we will want to copy and relocate the [entire module](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/image/paperclip_attachment.rb) to our application's directory to ensure proper functionality. This is because we will be telling Rails to ignore Spree's default `PaperclipAttachment` module.
+
+{% code title="app/models/amazing_store/spree/image/my_paperclip_attachment.rb" %}
+```ruby
+module AmazingStore::Spree::Image::MyPaperclipAttachment
+  extend ActiveSupport::Concern
+  # include the entire module
```

Comment:
What do you mean exactly with "include" here? Is it copy/paste?

Reply:
replaced with "copy/pasted module"